fbw-common/src/wasm/systems/systems/src/communications/amu.rs

```diff
@@ -0,0 +1,861 @@
+        }
+
+        writer.write(&self.volume_com1_id, self.mixed_audio.volume_com1);
+        writer.write(&self.volume_com2_id, self.mixed_audio.volume_com2);
+        writer.write(&self.volume_com3_id, self.mixed_audio.volume_com3);
+        writer.write(&self.volume_adf1_id, self.mixed_audio.volume_adf1);
+        writer.write(&self.volume_adf2_id, self.mixed_audio.volume_adf2);
+        writer.write(&self.volume_vor1_id, self.mixed_audio.volume_vor1);
+        writer.write(&self.volume_vor2_id, self.mixed_audio.volume_vor2);
+        writer.write(&self.volume_ils_id, self.mixed_audio.volume_ils);
+
+        // FOR FUTURE USE: Not needed for the time being as there's no K event for all this
+        // writer.write(&self.volume_markers_id, self.mixed_audio.volume_markers);
+        // writer.write(&self.volume_gls_id, self.mixed_audio.volume_gls);
+        // writer.write(&self.volume_hf1_id, self.volume_hf1);
+        // writer.write(&self.volume_hf2_id, self.volume_hf2);
+        // writer.write(&self.volume_att_id, self.volume_att);
+        // writer.write(&self.volume_mech_id, self.volume_mech);
+        // writer.write(&self.volume_pa_id, self.volume_pa);
+    }
+}
+
+pub struct Computer {
+    audio_card: AudioCard,
+    _second_card: TypeCard,
+
+    is_power_supply_powered: bool,
+}
+
+impl Computer {
+    pub fn new_cpt_and_avncs(context: &mut InitContext) -> Self {
+        Self {
+            audio_card: AudioCard::new(context, 1),
+            _second_card: TypeCard::Calls(Calls::new()),
+            is_power_supply_powered: false,
+        }
+    }
+
+    pub fn new_fo_and_ovhd(context: &mut InitContext) -> Self {
+        Self {
+            audio_card: AudioCard::new(context, 2),
+            // Not used for now
+            _second_card: TypeCard::Bite(Bite::new()),
+            is_power_supply_powered: false,
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, bus: &mut Vec<Arinc429Word<u32>>) {
+        self.audio_card
+            .update(context, bus, self.is_power_supply_powered);
+    }
+
+    pub fn get_mixed_audio_acp(&self) -> MixedAudio {
+        self.audio_card.get_mixed_audio_acp()
+    }
+
+    pub fn get_mixed_audio_acp3(&self) -> MixedAudio {
+        self.audio_card.get_mixed_audio_acp3()
+    }
+
+    pub fn get_transmission_table_acp(&self) -> u32 {
+        self.audio_card.get_transmission_table_acp()
+    }
+
+    pub fn get_transmission_table_acp3(&self) -> u32 {
+        self.audio_card.get_transmission_table_acp3()
+    }
+}
+
+impl SimulationElement for Computer {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.audio_card.accept(visitor);
+
+        visitor.visit(self);
+    }
+
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_power_supply_powered = buses.is_powered(ElectricalBusType::DirectCurrentEssential);
+    }
+}
+
+pub struct AudioCard {
```

Comment:
```suggestion
struct AudioCard {
```
Also remove `pub` on all the implemented functions.